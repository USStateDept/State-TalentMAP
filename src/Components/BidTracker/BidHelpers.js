import {
  APPROVED_PROP,
  CLOSED_PROP,
  DRAFT_PROP,
  HAND_SHAKE_ACCEPTED_PROP,
  HAND_SHAKE_OFFERED_PROP,
  HAND_SHAKE_DECLINED_PROP,
  PRE_PANEL_PROP,
  IN_PANEL_PROP,
  SUBMITTED_PROP,
  DECLINED_PROP,
  GET_DRAFT_TITLE,
  GET_SUBMIT_BID_ACTION_TITLE,
  GET_SUBMIT_BID_COMPLETE_TITLE,
  GET_HAND_SHAKE_EVALUATE_TITLE,
  GET_HAND_SHAKE_EVALUATING_TITLE,
  GET_HAND_SHAKE_OFFERED_TITLE,
  GET_HAND_SHAKE_ACCEPTED_TITLE,
  GET_PANEL_TITLE,
  GET_APPROVAL_TITLE,
  DRAFT_NUMBER,
  SUBMITTED_NUMBER,
  HAND_SHAKE_OFFERED_NUMBER,
  HAND_SHAKE_ACCEPTED_NUMBER,
  IN_PANEL_NUMBER,
  APPROVED_NUMBER,
} from '../../Constants/BidData';

// determine whether to show an alert on the bid tracker based on the status
export function shouldShowAlert(status) {
  const alertStatusArray = [HAND_SHAKE_OFFERED_PROP, IN_PANEL_PROP, APPROVED_PROP, CLOSED_PROP,
    HAND_SHAKE_DECLINED_PROP, DECLINED_PROP];

  if (alertStatusArray.indexOf(status) > -1) {
    return true;
  }
  return false;
}

// Determine what actions to allow as far as "deleting" and "withdrawing" based on the status.
// Also return a value to determine whether to show that action, but keep it disabled.
export function getActionPermissions(status) {
  const defaultPermissions = {
    showDelete: true,
    disableDelete: false,
    showWithdraw: false,
    disableWithdraw: false,
  };
  switch (status) {
    case DRAFT_PROP:
    case SUBMITTED_PROP:
      return defaultPermissions;
    case HAND_SHAKE_OFFERED_PROP:
    case HAND_SHAKE_ACCEPTED_PROP:
    case PRE_PANEL_PROP:
      return { ...defaultPermissions, showDelete: false, showWithdraw: true };
    default:
      return {
        ...defaultPermissions, showDelete: false, showWithdraw: true, disableWithdraw: true,
      };
  }
}

// Get an object with stages that map to six sections of the bid tracker.
// There's a prop for each stage (status), which contains data used to apply different styles.
// Instead of trying to apply some logic based on the current step, we explicitly
// declare what properties each step should have based on the status of the bid.
// This way, we can ensure a known, consistent output.
// The function accepts the entire bid object so that we can most importantly get the status,
// but also information such as dates so that if we want,
// we could dynamically render step titles within the function.
export function bidClassesFromCurrentStatus(bid = { status: 'draft' }) {
  // Configure defaults for a step that is incomplete (future stages).
  const DEFAULT_INCOMPLETE_OBJECT = {
    needsAction: false,
    isComplete: false,
    isCurrent: false,
    isPendingLine: false,
    isPendingAlert: false,
  };

  // Configure defaults for a complete object. The only thing we change is the isComplete prop.
  const DEFAULT_COMPLETE_OBJECT = { ...DEFAULT_INCOMPLETE_OBJECT, isComplete: true };

  // This is the base object we'll return. The stages prop will be filled with prop names
  // that correspond to each stage of the bid tracker.
  const bidClassObject = Object.assign({ stages: {} });

  const DRAFT_TITLE = GET_DRAFT_TITLE();
  const SUBMIT_BID_ACTION_TITLE = GET_SUBMIT_BID_ACTION_TITLE();
  const SUBMIT_BID_COMPLETE_TITLE = GET_SUBMIT_BID_COMPLETE_TITLE();
  const HAND_SHAKE_EVALUATE_TITLE = GET_HAND_SHAKE_EVALUATE_TITLE();
  const HAND_SHAKE_EVALUATING_TITLE = GET_HAND_SHAKE_EVALUATING_TITLE();
  const HAND_SHAKE_OFFERED_TITLE = GET_HAND_SHAKE_OFFERED_TITLE();
  const HAND_SHAKE_ACCEPTED_TITLE = GET_HAND_SHAKE_ACCEPTED_TITLE();
  const PANEL_TITLE = GET_PANEL_TITLE();
  const APPROVAL_TITLE = GET_APPROVAL_TITLE();

  const DRAFT_DATE = bid.draft_date;
  const SUBMITTED_DATE = bid.submitted_date;
  const HAND_SHAKE_OFFERED_DATE = bid.handshake_offered_date;
  const HAND_SHAKE_ACCEPTED_DATE = bid.handshake_accepted_date;
  // const HAND_SHAKE_DECLINED_DATE = bid.handshake_declined_date;
  const IN_PANEL_DATE = bid.in_panel_date;
  // const SCHEDULED_PANEL_DATE = bid.scheduled_panel_date;
  const APPROVED_DATE = bid.approved_date;
  // const DECLINED_DATE = bid.declined_date;
  // const CLOSED_DATE = bid.closed_date;
  // const CREATE_DATE = bid.create_date;
  // const UPDATE_DATE = bid.update_date;

  // Perform a switch to check the status.
  switch (bid.status) {

    // Draft stage
    case DRAFT_PROP:
      bidClassObject.stages[DRAFT_PROP] = {
        ...DEFAULT_COMPLETE_OBJECT, date: DRAFT_DATE, title: DRAFT_TITLE };
      bidClassObject.stages[SUBMITTED_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: SUBMITTED_DATE,
        needsAction: true,
        isCurrent: true,
        isPendingLine: true,
        title: SUBMIT_BID_ACTION_TITLE,
        number: SUBMITTED_NUMBER };
      bidClassObject.stages[HAND_SHAKE_OFFERED_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: HAND_SHAKE_OFFERED_DATE,
        title: HAND_SHAKE_EVALUATE_TITLE,
        number: HAND_SHAKE_OFFERED_NUMBER };
      bidClassObject.stages[HAND_SHAKE_ACCEPTED_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: HAND_SHAKE_ACCEPTED_DATE,
        title: HAND_SHAKE_ACCEPTED_TITLE,
        number: HAND_SHAKE_ACCEPTED_NUMBER };
      bidClassObject.stages[IN_PANEL_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: IN_PANEL_DATE,
        title: PANEL_TITLE,
        number: IN_PANEL_NUMBER };
      bidClassObject.stages[APPROVED_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: APPROVED_DATE,
        title: APPROVAL_TITLE,
        number: APPROVED_NUMBER };
      return bidClassObject;

    // Submitted stage, but we'll still render this
    // for handshake_offered, hand_shake_declined, and closed as well,
    // since we display an overlay alert for the user to accept the handshake,
    // or when they've declined it.
    case SUBMITTED_PROP:
    case HAND_SHAKE_OFFERED_PROP:
    case HAND_SHAKE_DECLINED_PROP:
    case CLOSED_PROP:
      bidClassObject.stages[DRAFT_PROP] = Object.assign(
        DEFAULT_COMPLETE_OBJECT,
        { number: DRAFT_NUMBER, date: DRAFT_DATE, title: DRAFT_TITLE },
      );
      bidClassObject.stages[SUBMITTED_PROP] = {
        ...DEFAULT_COMPLETE_OBJECT,
        date: SUBMITTED_DATE,
        title: SUBMIT_BID_COMPLETE_TITLE,
        number: SUBMITTED_NUMBER };
      bidClassObject.stages[HAND_SHAKE_OFFERED_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: HAND_SHAKE_OFFERED_DATE,
        title: HAND_SHAKE_EVALUATING_TITLE,
        needsAction: false,
        isCurrent: true,
        hasPendingTooltip: true,
        number: HAND_SHAKE_OFFERED_NUMBER };
      bidClassObject.stages[HAND_SHAKE_ACCEPTED_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: HAND_SHAKE_ACCEPTED_DATE,
        title: HAND_SHAKE_ACCEPTED_TITLE,
        number: HAND_SHAKE_ACCEPTED_NUMBER };
      bidClassObject.stages[IN_PANEL_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: IN_PANEL_DATE,
        title: PANEL_TITLE,
        number: IN_PANEL_NUMBER };
      bidClassObject.stages[APPROVED_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: APPROVED_DATE,
        title: APPROVAL_TITLE,
        number: APPROVED_NUMBER };
      return bidClassObject;

    // When the handshake is accepted, we display an overlay alert, so it can look the same
    // as the Pre-panel phase.
    case HAND_SHAKE_ACCEPTED_PROP:
    case PRE_PANEL_PROP:
      bidClassObject.stages[DRAFT_PROP] = Object.assign(
        DEFAULT_COMPLETE_OBJECT,
        { number: DRAFT_NUMBER, date: DRAFT_DATE, title: DRAFT_TITLE },
      );
      bidClassObject.stages[SUBMITTED_PROP] = {
        ...DEFAULT_COMPLETE_OBJECT,
        date: SUBMITTED_DATE,
        title: SUBMIT_BID_COMPLETE_TITLE,
        number: SUBMITTED_NUMBER };
      bidClassObject.stages[HAND_SHAKE_OFFERED_PROP] = {
        ...DEFAULT_COMPLETE_OBJECT,
        date: HAND_SHAKE_OFFERED_DATE,
        title: HAND_SHAKE_OFFERED_TITLE,
        number: SUBMITTED_NUMBER };
      bidClassObject.stages[HAND_SHAKE_ACCEPTED_PROP] = {
        ...DEFAULT_COMPLETE_OBJECT,
        date: HAND_SHAKE_ACCEPTED_DATE,
        title: HAND_SHAKE_ACCEPTED_TITLE,
        needsAction: false,
        isPendingLine: true,
        number: HAND_SHAKE_ACCEPTED_NUMBER };
      bidClassObject.stages[IN_PANEL_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: IN_PANEL_DATE,
        title: PANEL_TITLE,
        number: IN_PANEL_NUMBER };
      bidClassObject.stages[APPROVED_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: APPROVED_DATE,
        title: APPROVAL_TITLE,
        number: APPROVED_NUMBER };
      return bidClassObject;

    // When the bid is approved, we display an overlay alert, so we can render
    // the tracker the same as the in-panel phase.
    case IN_PANEL_PROP:
    case APPROVED_PROP:
    case DECLINED_PROP:
      bidClassObject.stages[DRAFT_PROP] = Object.assign(
        DEFAULT_COMPLETE_OBJECT,
        { number: DRAFT_NUMBER, date: DRAFT_DATE, title: DRAFT_TITLE },
      );
      bidClassObject.stages[SUBMITTED_PROP] = {
        ...DEFAULT_COMPLETE_OBJECT,
        date: SUBMITTED_DATE,
        title: SUBMIT_BID_COMPLETE_TITLE,
        number: SUBMITTED_NUMBER };
      bidClassObject.stages[HAND_SHAKE_OFFERED_PROP] = {
        ...DEFAULT_COMPLETE_OBJECT,
        date: HAND_SHAKE_OFFERED_DATE,
        title: HAND_SHAKE_OFFERED_TITLE,
        number: SUBMITTED_NUMBER };
      bidClassObject.stages[HAND_SHAKE_ACCEPTED_PROP] = {
        ...DEFAULT_COMPLETE_OBJECT,
        date: HAND_SHAKE_ACCEPTED_DATE,
        title: HAND_SHAKE_ACCEPTED_TITLE,
        number: HAND_SHAKE_ACCEPTED_NUMBER };
      bidClassObject.stages[IN_PANEL_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: IN_PANEL_DATE,
        title: PANEL_TITLE,
        needsAction: false,
        isCurrent: true,
        number: IN_PANEL_NUMBER };
      bidClassObject.stages[APPROVED_PROP] = {
        ...DEFAULT_INCOMPLETE_OBJECT,
        date: APPROVED_DATE,
        title: APPROVAL_TITLE,
        number: APPROVED_NUMBER };
      return bidClassObject;
    default:
      return false;
  }
}
