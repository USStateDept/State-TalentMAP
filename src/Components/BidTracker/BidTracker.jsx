import React from 'react';
import PropTypes from 'prop-types';
import { BID_LIST, NOTIFICATION_LIST, USER_PROFILE } from '../../Constants/PropTypes';
import BidTrackerCardList from './BidTrackerCardList';
import Spinner from '../Spinner';
import NotificationsSection from './NotificationsSection';
import ContactCDOButton from './ContactCDOButton';

const BidTracker = ({ bidList, bidListIsLoading, acceptBid, declineBid,
notifications, notificationsIsLoading, markBidTrackerNotification, userProfile,
userProfileIsLoading }) => (
  <div className="bid-tracker-page">
    <NotificationsSection
      notifications={notifications}
      notificationsIsLoading={notificationsIsLoading}
      markBidTrackerNotification={markBidTrackerNotification}
    />
    <div className="usa-grid-full">
      <div className="usa-width-one-half bid-tracker-greeting-container">
        <div className="hello-greeting bid-tracker-greeting">
          Bid Tracker
        </div>
      </div>
      <div className="usa-width-one-half bid-tracker-cdo-email-container">
        <div className="bid-tracker-cdo-email">
          {
            userProfile.cdo && !userProfileIsLoading &&
            <ContactCDOButton email={userProfile.cdo.email} />
          }
        </div>
      </div>
    </div>
    {
      bidListIsLoading ?
        <Spinner type="homepage-position-results" size="big" /> :
        <div className="usa-grid-full">
          <BidTrackerCardList
            bids={bidList.results}
            acceptBid={acceptBid}
            declineBid={declineBid}
          />
        </div>
    }
  </div>
);

BidTracker.propTypes = {
  bidList: BID_LIST.isRequired,
  bidListIsLoading: PropTypes.bool.isRequired,
  acceptBid: PropTypes.func.isRequired,
  declineBid: PropTypes.func.isRequired,
  notifications: NOTIFICATION_LIST.isRequired,
  notificationsIsLoading: PropTypes.bool.isRequired,
  markBidTrackerNotification: PropTypes.func.isRequired,
  userProfile: USER_PROFILE.isRequired,
  userProfileIsLoading: PropTypes.bool.isRequired,
};

export default BidTracker;
