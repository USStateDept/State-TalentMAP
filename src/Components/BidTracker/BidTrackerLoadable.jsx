import createLoader from '../Loadable';

export const path = () => import('./BidTracker');

const BidTracker = createLoader({ path, shouldPreload: false });

const BidTrackerLoadable = ({ ...rest }) => (
  <BidTracker {...rest} />
);

export default BidTrackerLoadable;
