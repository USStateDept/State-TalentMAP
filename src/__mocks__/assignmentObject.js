const assignmentObject = {
  start_date: '10-10-2016',
  end_date: '10-10-2018',
  estimated_end_date: '10-10-2018',
  position: {
    skill: 'CONSTRUCTION ENGINEERING (6218)',
    bureau: '(NEA) BUREAU NEAR EASTERN AFFAIRS',
    position_number: '58348065',
    title: 'PROJECT DIRECTOR',
    post: {
      location: {
        id: 4,
        country: 'United States',
        code: '500',
        city: 'Washington',
        state: 'DC',
      },
    },
  },
};

export default assignmentObject;
