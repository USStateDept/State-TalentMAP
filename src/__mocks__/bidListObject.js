const bidListObject = {
  count: 3,
  next: null,
  previous: null,
  results: [
    {
      id: 3,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 226,
        grade: '03',
        skill: 'POLITICAL AND ECONOMIC AFFAIRS (6050)',
        position_number: '10035561',
        title: 'POLITICAL/ECONOMIC OFFICER',
        create_date: '2015-08-05',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 199,
          location: 'Curacao',
        },
      },
      status: 'approved',
      submission_date: '2017-10-31',
      update_date: '2017-10-31',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 1,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 1,
        grade: '05',
        skill: 'OFFICE MANAGEMENT (9017)',
        position_number: '00003026',
        title: 'OMS (COM)',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 235,
          location: 'Freetown, Sierra Leone',
        },
      },
      status: 'closed',
      submission_date: '2017-10-31',
      update_date: '2017-11-01',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 2,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 84,
        grade: '04',
        skill: 'SECURITY (2501)',
        position_number: '56001600',
        title: 'SPECIAL AGENT',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 195,
          location: 'The Hague, Netherlands',
        },
      },
      status: 'closed',
      submission_date: null,
      update_date: '2017-11-01',
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
  ],
};

export default bidListObject;
