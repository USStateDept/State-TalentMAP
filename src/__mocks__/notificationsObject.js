const notificationsObject = {
  count: 4,
  results: [
    {
      id: 1,
      owner: 'townpostj',
      message: 'Saved search New BidCycle has 13 new results available',
      is_read: false,
      date_created: '2017-10-31T18:14:07.262935Z',
      date_updated: '2017-10-31T18:14:07.262965Z',
      tags: [
        'waiver',
      ],
    },
    {
      id: 6,
      owner: 'townpostj',
      message: 'Your bid for [10035561] POLITICAL/ECONOMIC OFFICER (Curacao) has been offered a handshake.',
      is_read: false,
      date_created: '2017-11-01T14:56:55.508891Z',
      date_updated: '2017-11-01T14:56:55.508920Z',
      tags: [
        'bidding',
      ],
    },
    {
      id: 8,
      owner: 'townpostj',
      message: 'Your bid for [10035561] POLITICAL/ECONOMIC OFFICER (Curacao) is under panel review.',
      is_read: true,
      date_created: '2017-11-01T14:57:10.189772Z',
      date_updated: '2017-11-01T14:57:10.189803Z',
      tags: [
        'waiver',
      ],
    },
    {
      id: 9,
      owner: 'townpostj',
      message: 'Your bid for [10035561] POLITICAL/ECONOMIC OFFICER (Curacao) has been approved by panel.',
      is_read: false,
      date_created: '2017-11-01T14:57:17.257837Z',
      date_updated: '2017-11-01T14:57:17.257866Z',
      tags: [
        'bidding',
      ],
    },
  ],
};

export default notificationsObject;
