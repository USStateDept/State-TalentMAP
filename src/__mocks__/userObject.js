export const bidderUserObject = {
  skill_code: ['INFORMATION MANAGEMENT (2880)', 'ADMINISTRATIVE SUPPORT (2080)'],
  user: {
    first_name: 'John',
    last_name: 'Doe',
    email: 'doej@state.gov',
    username: 'jdoe',
  },
  is_cdo: false,
  cdo: {
    email: 'shadtrachl@state.gov',
  },
  bid_statistics: [
    {
      draft: 1,
      submitted: 1,
    },
  ],
};

export const cdoUserObject = Object.assign({}, bidderUserObject, { is_cdo: true });
