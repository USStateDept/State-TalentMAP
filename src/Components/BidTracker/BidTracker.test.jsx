import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import BidTracker from './BidTracker';
import bidListObject from '../../__mocks__/bidListObject';
import notificationsObject from '../../__mocks__/notificationsObject';
import { bidderUserObject } from '../../__mocks__/userObject';

describe('BidTrackerComponent', () => {
  const props = {
    bidList: bidListObject,
    bidListIsLoading: false,
    acceptBid: () => {},
    declineBid: () => {},
    notifications: notificationsObject,
    notificationsIsLoading: false,
    markBidTrackerNotification: () => {},
    userProfile: bidderUserObject,
    userProfileIsLoading: false,
  };

  it('is defined', () => {
    const wrapper = shallow(
      <BidTracker {...props} />,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <BidTracker {...props} />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when bidListIsLoading is true', () => {
    const wrapper = shallow(
      <BidTracker {...props} bidListIsLoading />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
