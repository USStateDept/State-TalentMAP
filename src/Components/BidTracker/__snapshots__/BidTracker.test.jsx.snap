// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidTrackerComponent matches snapshot 1`] = `
<div
  className="bid-tracker-page"
>
  <NotificationsSection
    markBidTrackerNotification={[Function]}
    notifications={
      Object {
        "count": 4,
        "results": Array [
          Object {
            "date_created": "2017-10-31T18:14:07.262935Z",
            "date_updated": "2017-10-31T18:14:07.262965Z",
            "id": 1,
            "is_read": false,
            "message": "Saved search New BidCycle has 13 new results available",
            "owner": "townpostj",
            "tags": Array [
              "waiver",
            ],
          },
          Object {
            "date_created": "2017-11-01T14:56:55.508891Z",
            "date_updated": "2017-11-01T14:56:55.508920Z",
            "id": 6,
            "is_read": false,
            "message": "Your bid for [10035561] POLITICAL/ECONOMIC OFFICER (Curacao) has been offered a handshake.",
            "owner": "townpostj",
            "tags": Array [
              "bidding",
            ],
          },
          Object {
            "date_created": "2017-11-01T14:57:10.189772Z",
            "date_updated": "2017-11-01T14:57:10.189803Z",
            "id": 8,
            "is_read": true,
            "message": "Your bid for [10035561] POLITICAL/ECONOMIC OFFICER (Curacao) is under panel review.",
            "owner": "townpostj",
            "tags": Array [
              "waiver",
            ],
          },
          Object {
            "date_created": "2017-11-01T14:57:17.257837Z",
            "date_updated": "2017-11-01T14:57:17.257866Z",
            "id": 9,
            "is_read": false,
            "message": "Your bid for [10035561] POLITICAL/ECONOMIC OFFICER (Curacao) has been approved by panel.",
            "owner": "townpostj",
            "tags": Array [
              "bidding",
            ],
          },
        ],
      }
    }
    notificationsIsLoading={false}
  />
  <div
    className="usa-grid-full"
  >
    <div
      className="usa-width-one-half bid-tracker-greeting-container"
    >
      <div
        className="hello-greeting bid-tracker-greeting"
      >
        Bid Tracker
      </div>
    </div>
    <div
      className="usa-width-one-half bid-tracker-cdo-email-container"
    >
      <div
        className="bid-tracker-cdo-email"
      >
        <ContactCDOButton
          email="shadtrachl@state.gov"
        />
      </div>
    </div>
  </div>
  <div
    className="usa-grid-full"
  >
    <BidTrackerCardList
      acceptBid={[Function]}
      bids={
        Array [
          Object {
            "bidcycle": "Demo BidCycle 2017-10-31 16:00:51.254762",
            "id": 3,
            "position": Object {
              "bureau": "(AF) BUREAU OF AFRICAN AFFAIRS",
              "create_date": "2015-08-05",
              "grade": "03",
              "id": 226,
              "position_number": "10035561",
              "post": Object {
                "id": 199,
                "location": "Curacao",
              },
              "skill": "POLITICAL AND ECONOMIC AFFAIRS (6050)",
              "title": "POLITICAL/ECONOMIC OFFICER",
              "update_date": "2017-06-08",
            },
            "reviewer": Object {
              "email": "woodwardw@state.gov",
              "first_name": "Wendy",
              "is_cdo": false,
              "last_name": "Woodward",
              "phone_number": "555-555-5555",
              "username": "woodwardw",
            },
            "status": "approved",
            "submission_date": "2017-10-31",
            "update_date": "2017-10-31",
            "user": "rehmant",
          },
          Object {
            "bidcycle": "Demo BidCycle 2017-10-31 16:00:51.254762",
            "id": 1,
            "position": Object {
              "bureau": "(AF) BUREAU OF AFRICAN AFFAIRS",
              "create_date": "2006-09-20",
              "grade": "05",
              "id": 1,
              "position_number": "00003026",
              "post": Object {
                "id": 235,
                "location": "Freetown, Sierra Leone",
              },
              "skill": "OFFICE MANAGEMENT (9017)",
              "title": "OMS (COM)",
              "update_date": "2017-06-08",
            },
            "reviewer": Object {
              "email": "woodwardw@state.gov",
              "first_name": "Wendy",
              "is_cdo": false,
              "last_name": "Woodward",
              "phone_number": "555-555-5555",
              "username": "woodwardw",
            },
            "status": "closed",
            "submission_date": "2017-10-31",
            "update_date": "2017-11-01",
            "user": "rehmant",
          },
          Object {
            "bidcycle": "Demo BidCycle 2017-10-31 16:00:51.254762",
            "id": 2,
            "position": Object {
              "bureau": "(AF) BUREAU OF AFRICAN AFFAIRS",
              "create_date": "2006-09-20",
              "grade": "04",
              "id": 84,
              "position_number": "56001600",
              "post": Object {
                "id": 195,
                "location": "The Hague, Netherlands",
              },
              "skill": "SECURITY (2501)",
              "title": "SPECIAL AGENT",
              "update_date": "2017-06-08",
            },
            "reviewer": Object {
              "email": "woodwardw@state.gov",
              "first_name": "Wendy",
              "is_cdo": false,
              "last_name": "Woodward",
              "phone_number": "555-555-5555",
              "username": "woodwardw",
            },
            "status": "closed",
            "submission_date": null,
            "update_date": "2017-11-01",
            "user": "rehmant",
          },
        ]
      }
      declineBid={[Function]}
    />
  </div>
</div>
`;

exports[`BidTrackerComponent matches snapshot when bidListIsLoading is true 1`] = `
<div
  className="bid-tracker-page"
>
  <NotificationsSection
    markBidTrackerNotification={[Function]}
    notifications={
      Object {
        "count": 4,
        "results": Array [
          Object {
            "date_created": "2017-10-31T18:14:07.262935Z",
            "date_updated": "2017-10-31T18:14:07.262965Z",
            "id": 1,
            "is_read": false,
            "message": "Saved search New BidCycle has 13 new results available",
            "owner": "townpostj",
            "tags": Array [
              "waiver",
            ],
          },
          Object {
            "date_created": "2017-11-01T14:56:55.508891Z",
            "date_updated": "2017-11-01T14:56:55.508920Z",
            "id": 6,
            "is_read": false,
            "message": "Your bid for [10035561] POLITICAL/ECONOMIC OFFICER (Curacao) has been offered a handshake.",
            "owner": "townpostj",
            "tags": Array [
              "bidding",
            ],
          },
          Object {
            "date_created": "2017-11-01T14:57:10.189772Z",
            "date_updated": "2017-11-01T14:57:10.189803Z",
            "id": 8,
            "is_read": true,
            "message": "Your bid for [10035561] POLITICAL/ECONOMIC OFFICER (Curacao) is under panel review.",
            "owner": "townpostj",
            "tags": Array [
              "waiver",
            ],
          },
          Object {
            "date_created": "2017-11-01T14:57:17.257837Z",
            "date_updated": "2017-11-01T14:57:17.257866Z",
            "id": 9,
            "is_read": false,
            "message": "Your bid for [10035561] POLITICAL/ECONOMIC OFFICER (Curacao) has been approved by panel.",
            "owner": "townpostj",
            "tags": Array [
              "bidding",
            ],
          },
        ],
      }
    }
    notificationsIsLoading={false}
  />
  <div
    className="usa-grid-full"
  >
    <div
      className="usa-width-one-half bid-tracker-greeting-container"
    >
      <div
        className="hello-greeting bid-tracker-greeting"
      >
        Bid Tracker
      </div>
    </div>
    <div
      className="usa-width-one-half bid-tracker-cdo-email-container"
    >
      <div
        className="bid-tracker-cdo-email"
      >
        <ContactCDOButton
          email="shadtrachl@state.gov"
        />
      </div>
    </div>
  </div>
  <Spinner
    filled={false}
    inverse={false}
    size="big"
    type="homepage-position-results"
  />
</div>
`;
